src/tasks/task.ts

```diff
@@ -36,7 +36,7 @@ import { TaskBase } from "./taskinterfaces";
  * return firstDone.result;
  * ```
  */
-export class Task implements TaskBase {
+export class Task<T = unknown> implements TaskBase {

```

Comment:
I'm not sure I see the benefit of using generic parameters for `Task` unless we can enforce `yield` on `Task<T>` returns a value of T.

Reply:
Yeah, as I mentioned up top, the use of `yield` leaves you stuck with type inference unless https://github.com/microsoft/TypeScript/issues/36967 is resolved. The only alternative would be to replace the usage of `Task` with a `Promise` and move away from generator functions to Promise-based async, but that'd be a rather huge change that I probably wouldn't advise without a major version bump and breaking changes 😛

Reply:
100% agree. It would be really nice to get better type safety, but it's already somewhat user-enforced in C# even due to the fact that activities and suborchestrations are referenced by name instead of by type.